connectors/byoc.py

```diff
@@ -385,6 +386,7 @@ async def sync(self, data_provider, elastic_server, idling, sync_now=False):
                         await self.sync_doc()
                     return
             else:
+                self.sync_now = self.doc_source["sync_now"] = True
```

Comment:
I do this to avoid passing one more param. Or I can make it explicit, `sync_now_from_cli`.

Reply:
no, that way of writing is great. I just want to say that when you update `self.doc_source` it's not specifically updated on the server side (that's done with `_write` and `save`)

Reply:
> I just want to say that when you update `self.doc_source` it's not specifically updated on the server side (that's done with `sync_doc`)

It shouldn't.

Let's say you run the connector service with
```shell
bin/elastic-ingest --action poll --sync-now  
```
which will force a job for each connector the service can run. But I don't think it needs to update `sync_now` flag of each connector document in `.elastic-connectors`.

Or do you mean we should always keep `self.sync_now` and `self.doc_source["sync_now"]` in sync?

Reply:
if you change `self.doc_source` it will be updated in Elasticsearch, and that field is not in the read only list of fields, so that value will be updated. So it seems that you don't want to change it, so it should be removed or the field added in the read only fields 

I know this is cumbersome and not clear, but I talked with some others in the team to have a clearer state representation if the doc and avoid duplicating flags locally like that.

Reply:
> how is this done?

everytime `sync()` is called

> This is not an option as we need to reset the flag after a job is started

but if you don't write back the value to Elasticsearch, why do you want to change `self.doc_source["sync_now"]` ?

Reply:
>everytime sync() is called

I believe the data is persisted by this method:
https://github.com/elastic/connectors-python/blob/7346aae53adeabc3c824bb1e0dfcc3b582c7ecab/connectors/byoc.py#L76

>but if you don't write back the value to Elasticsearch, why do you want to change self.doc_source["sync_now"] ?

Good point. I have changed it to only update `self.sync_now`. Based on the implementation of `sync()` method above, it will persist data based on the value of `self.doc_source`.